src/features/profile/Profile.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useSelector } from "react-redux";
+import useProfileState from "./use-profile-state";
+import { AuthState } from "features/auth/store/types";
+import { fetchOneProfile } from "./services/profile.service";
+import ShowSkills from "./components/show-skills";
+import ShowExperiences from "./components/show-experiences";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+
+const Profile = () => {
+  const { profile, setProfile } = useProfileState();
+
+  const id = parseInt(useParams<RouteParams>().id!);
+
+  function useAuthState(): AuthState {
+    return useSelector<{ auth: AuthState }>((state) => state.auth) as AuthState;
+  }
+  const { token } = useAuthState();
+  const loggedUser = useAuthState().user;
+
+  const getProfile = useCallback(async () => {
+    try {
+      if (token && id) {
+        const data = await fetchOneProfile(token, id);
+        setProfile(data);
+        console.log(token);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }, [token, id, setProfile]);
+
+  useEffect(() => {
+    getProfile();
+  }, [getProfile]);
```

Comment:
el `useCallback getProfile` y el `useEffect` que lo activa, perfectamente pueden estar dentro de `useProfileState()`; eso reduciria el codigo aqui, haria que `useProfileState` tuviera mas cohesion y ademas haria que evitemos exponer `setProfile` ya que, ¿para que lo queremos sacar?, haria que fuese un estado de "read-only" para lo que respecta al componente

```suggestion
  const profile = useProfileState();

  const id = parseInt(useParams<RouteParams>().id!);

  function useAuthState(): AuthState {
    return useSelector<{ auth: AuthState }>((state) => state.auth) as AuthState;
  }
  const { token } = useAuthState();
  const loggedUser = useAuthState().user;
```

Aun asi, tambien podemos exponer quiza "getProfile" junto a "profile" por si lo queremos usar para re-cargar los datos del perfil (lo cual es bastante comun)

Si hacemos esto, lo que comente en mi comentario anterior de que ya no es necesario encapsular si vamos por la opcion numero 2, seria falso, ahora si hay razon para encapsular

Reply:
creo que queda mas bonito colocandolo en otro componente, tambien lo pense pero no sabia que tan bueno era hacerlo XD

Reply:
Hmm puede ir en otro componente tambien

Pero ese seria un componente padre que tendria dentro de si ese custom-hook del que te estoy hablando para despues inyectarlo por medio de props al componente hijo que seria este

Esta bien hacerlo pero no veo necesidad de ello 🤔, lo que si veo necesidad es de separarlo en un custom-hook que sea lo hagas en el hijo o en el padre lo tienes que hacer

Por que?

Por que hay muchos hooks, variables y propiedades dentro de ese custom-hook que te estoy comentando que realmente no le importan ni al padre ni al hijo, por eso conviene encapsularlos, en persona te podria explicar un poco mas de esta logica por que es fundamental no solo para custom-hooks si no para encapsulamiento en general de CUALQUIER COSA xP